compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp

```diff
@@ -76,6 +76,33 @@ luci::CircleReshape *create_cloned_reshape(luci::CircleReshape *reshape)
   return new_reshape;
 }
 
+bool forward_reshape(luci::CircleReshape *reshape, luci::CircleMean *mean, uint32_t axis)
+{
+  assert(reshape != nullptr); // FIX_CALLER_UNLESS
+  assert(mean != nullptr);    // FIX_CALLER_UNLESS
+
+  auto new_reshape = create_cloned_reshape(reshape);
+  if (not new_reshape)
+    return false;
+
+  // reconnect network
+  loco::replace(mean).with(new_reshape);
+  mean->input(reshape->tensor());
+  new_reshape->tensor(mean);
+
+  // Change const shape axis value
+  auto *shape_reshape = loco::must_cast<luci::CircleConst *>(new_reshape->shape());
+  assert(shape_reshape->dtype() == loco::DataType::S32);     // FIX_CALLER_UNLESS
+  assert(axis < shape_reshape->size<loco::DataType::S32>()); // FIX_CALLER_UNLESS
+  shape_reshape->at<loco::DataType::S32>(axis) = 1;
```

Comment:
Q) why put `1` ?

Reply:
Since we are swapping the reshape and mean operation, then after applying mean first, the axis tensor dimension values will be equal to 1, and accordingly it is necessary to change the reshape_const value so that it corresponds to the new input for reshape.

Reply:
Thanks for the long explanation, but I don't understand (sorry for my ignorance) why it is equal to 1. Can you give more information about this?

Reply:
@BalyshevArtem Forwarding reshape over mean seems not trivial. Could you add a test as you did in `FuseMulWithDivPass` ?  It would be good to draw BEFORE and AFTER graph for readers.

Reply:
> Could you add a test as you did in `FuseMulWithDivPass` ? It would be good to draw BEFORE and AFTER graph for readers.

Added test.


> Can you give more information about this?

after the mean operation on the axis dimension, the value will always be equal to one. Therefore, shape const should also change its axis value by one:
```
/**
 *  Simple graph for test
 *
 *  BEFORE
 *               [Input]
 *              (3, 4, 4)                 [Shape_Const = (1, -1, 4)]
 *                  |                               |
 *              [Reshape] ----------------
 *              (1, 12, 4)
 *                  |
 *        [Mean, keep_dims = true]
 *              (1, 12, 1)
 *                  |
 *               [Output]
 *
 *  AFTER
 *               [Input]
 *              (3, 4, 4)
 *                  |
 *         [Mean, keep_dims = true]
 *              (3, 4, 1)                 [Shape_Const = (1, -1, 1)]
 *                  |                               |
 *              [Reshape]-----------------
 *              (1, 12, 1)
 *                  |
 *              [Output]
 *
 */
```

Reply:
Thanks for the explanation! Now I understand :)

Reply:
It would be nice to put BEFORE/AFTER diagram in the source

Reply:
Found graph in test code. let's move that to here.

Reply:
> Found graph in test code. let's move that to here.

Done